mitodl/ocw-hugo-themes
# Subscripts and Superscripts

#### Pre-Flight checklist


#### What are the relevant tickets?

https://github.com/mitodl/ocw-studio/issues/1061

#### What's this PR do?
This PR:
1. Makes a 2-line change to the resource-link shortcode, enabling HTML in its title content
2. Adds some tests around subscripts and superscripts.

#### How should this be manually tested?
1. This PR requires the hugo configs to allow HTML in the goldmark rendering. Use configs from ocw-hugo-projects branch https://github.com/mitodl/ocw-hugo-projects/pull/234
2. Run `yarn test:e2e` to verify the e2e tests pass. *They won't pass on CI until the hugo-projects PR is merged.*
3. Run `yarn test:e2e --debug` and visit the subscript/superscript zoo: http://localhost:3010/courses/ocw-ci-test-course/pages/subscripts-and-superscripts
4. Test content you created:
    1. Run ocw-studio as on branch https://github.com/mitodl/ocw-studio/pull/1588 as directed in that PR. (i.e., with the ocw-studio config from m/mitodl/ocw-hugo-projects/pull/234)
    2. Set env variables:
        ```
        OCW_HUGO_PROJECTS_BRANCH=cc/subsup
        OCW_HUGO_THEMES_BRANCH=cc/subsup
        ```
    3.  Run `docker compose exec web ./manage.py backpopulate_pipelines --filter=<whatever-course-name>` to repopulate the populishing pipeline with new env vars
    4. Make some subscripts and publish the course locally. *Note: Depending on the course you use, you made need to remove instructors  from the metadata, unless you have a local version of www published.*
    5. Check that the output looks correct.


#### Any background context you want to provide?
HQ discussion: https://github.com/mitodl/hq/discussions/303

#### Screenshots (if appropriate)

<img width="683" alt="Screen Shot 2022-11-30 at 9 00 36 AM" src="https://user-images.githubusercontent.com/9010790/204815653-319dcc45-ba90-4c52-a370-063374ac5965.png">
